### PR TITLE
manifest: openthread: openthread upmerge to `242c7cc`

### PR DESCRIPTION
**Pull request by canisLupus1313**
Regular openthread upmerge.

### CODE REVIEW

**Review by pdgendt — DISMISSED** [review]

File: modules/openthread/CMakeLists.txt [hunk 1]
```diff
@@ -446,6 +446,7 @@ if(CONFIG_OPENTHREAD_COPROCESSOR_VENDOR_HOOK_SOURCE)
   set(OT_NCP_VENDOR_HOOK_SOURCE ${CONFIG_OPENTHREAD_COPROCESSOR_VENDOR_HOOK_SOURCE} CACHE STRING "NCP vendor hook source file name" FORCE)
 endif()
 
+set(BUILD_TESTING OFF CACHE BOOL "Disable openthread cmake testing targets" FORCE)
```
Comment (edmont) [inline modules/openthread/CMakeLists.txt:449]: nit: add an empty line after this one.

**Review by edmont — DISMISSED** [review]

File: modules/openthread/CMakeLists.txt (hunk 1 above)
Comment (canisLupus1313) [reply modules/openthread/CMakeLists.txt:449]: fixed.

**Review by edmont — DISMISSED** [review]

File: west.yml [hunk 2]
```diff
@@ -203,7 +203,7 @@ manifest:
       revision: aedcc262f93bbb1b0c2f58026911575729b7465c
       path: modules/lib/open-amp
     - name: openthread
-      revision: a1979fdd8ed756468719cc4921b04a57efcf9f15
+      revision: pull/78/head
```
Comment (rlubos) [inline west.yml:206]: Please update SHA: `242c7cc7c3a80c985476689b64cdb6d8fbcf5b6d`

Comment (canisLupus1313) [reply west.yml]: done

**Review by rlubos — APPROVED** [review]

**Review by edmont — APPROVED** [review]

**Review by pdgendt — APPROVED** [review]

**Pull request merged**